bind-contrib/contrib/updatehosts-1.1.0/man/readinfo_pp.1
.\" $Id: readinfo_pp.1,v 1.1 2000/07/11 06:53:22 vixie Exp $
.TH READINFO_PP 1 $Date: 2000/07/11 06:53:22 $
.UC 4
.SH NAME
readinfo_pp \- pretty print a readinfo data file
.SH SYNOPSIS
.B readinfo
.SH DESCRIPTION
.PP
.B readinfo_pp
is a utility to pretty print a
.B readinfo
data file.
The program attempts to align columns in a readable format.
The input data file is read from stdin and the output is on stdout.
.SH
See also
.PP
.B readinfo(1)


CHANGES
5500.	[bug]		Fake entry.

5494.	[bug]		Silence the EPROTO syslog message on older systems.
			[GL #1928]

5493.	[bug]		Fix off-by-one error when calculating new hashtable
			size. [GL #2104]

5492.	[bug]		Tighten LOC parsing to reject period and/or m as a
			value.  Correct handling of negative altitudes which
			are not whole metres.  [GL #2074]

5491.	[bug]		rbtversion->glue_table_size could be read without the
			appropriate lock being held. [GL #2080]

5490.	[func]		Refactor the readline support to use pkg-config and
			add support for editline library. [GL !3942]

5489.	[bug]		Named failed to reject some invalid records resulting
			in records that, after being printed, could not be
			loaded or would result in DNSSEC validation failures
			when re-read from zone files as the wire format
			differed.  The covered records records are: CERT,
			IPSECKEY, NSEC3, NSEC3PARAM, NXT, SIG, TLSA, WKS, and
			X25. [GL !3953]

5488.	[bug]		nta needed to have a weak reference on view to prevent
			the view being deleted while nta tests are being
			performed. [GL #2067]

5487.	[cleanup]	Update managed keys log messages to be less confusing.
			[GL #2027]

5486.	[func]		Add 'rndc dnssec -checkds' command to tell named
			that the DS record has been published in the parent.
			[GL #1613]

	--- 9.17.4 released ---

5485.	[placeholder]

5484.	[func]		Expire zero TTL records quickly rather than using them
			for stale answers. [GL #1829]

5483.	[func]		Keeping "stale" answers in cache has been disabled by
			default and can be re-enabled with a new configuration
			option "stale-cache-enable". [GL #1712]

5482.	[bug]		If the Duplicate Address Detection (DAD) mechanism had
			not yet finished after adding a new IPv6 address to the
			system, BIND 9 would fail to bind to IPv6 addresses in a
			tentative state. [GL #2038]

5481.	[security]	"update-policy" rules of type "subdomain" were
			incorrectly treated as "zonesub" rules, which allowed
			keys used in "subdomain" rules to update names outside
			of the specified subdomains. The problem was fixed by
			making sure "subdomain" rules are again processed as
			described in the ARM. (CVE-2020-8624) [GL #2055]

5480.	[security]	When BIND 9 was compiled with native PKCS#11 support, it
			was possible to trigger an assertion failure in code
			determining the number of bits in the PKCS#11 RSA public
			key with a specially crafted packet. (CVE-2020-8623)
			[GL #2037]

5479.	[security]	named could crash in certain query resolution scenarios
			where QNAME minimization and forwarding were both
			enabled. (CVE-2020-8621) [GL #1997]

5478.	[security]	It was possible to trigger an assertion failure by
			sending a specially crafted large TCP DNS message.
			(CVE-2020-8620) [GL #1996]

5477.	[bug]		The idle timeout for connected TCP sockets, which was
			previously set to a high fixed value, is now derived
			from the client query processing timeout configured for
			a resolver. [GL #2024]

5476.	[security]	It was possible to trigger an assertion failure when
			verifying the response to a TSIG-signed request.
			(CVE-2020-8622) [GL #2028]

5475.	[bug]		Wildcard RPZ passthru rules could incorrectly be
			overridden by other rules that were loaded from RPZ
			zones which appeared later in the "response-policy"
			statement. This has been fixed. [GL #1619]

5474.	[bug]		dns_rdata_hip_next() failed to return ISC_R_NOMORE
			when it should have. [GL !3880]

5473.	[func]		The RBT hash table implementation has been changed
			to use a faster hash function (HalfSipHash2-4) and
			Fibonacci hashing for better distribution. Setting
			"max-cache-size" now preallocates a fixed-size hash
			table so that rehashing does not cause resolution
			brownouts while the hash table is grown. [GL #1775]

5472.	[func]		The statistics channel has been updated to use the
			new network manager. [GL #2022]

5471.	[bug]		The introduction of KASP support inadvertently caused
			the second field of "sig-validity-interval" to always be
			calculated in hours, even in cases when it should have
			been calculated in days. This has been fixed. (Thanks to
			Tony Finch.) [GL !3735]

5470.	[port]		gsskrb5_register_acceptor_identity() is now only called
			if gssapi_krb5.h is present. [GL #1995]

5469.	[port]		On illumos, a constant called SEC is already defined in
			<sys/time.h>, which conflicts with an identically named
			constant in libbind9. This conflict has been resolved.
			[GL #1993]

5468.	[bug]		Addressed potential double unlock in process_fd().
			[GL #2005]

5467.	[func]		The control channel and the rndc utility have been
			updated to use the new network manager. To support
			this, the network manager was updated to enable
			the initiation of client TCP connections. Its
			internal reference counting has been refactored.

			Note: As a side effect of this change, rndc cannot
			currently be used with UNIX-domain sockets, and its
			default timeout has changed from 60 seconds to 30.
			These will be addressed in a future release.
			[GL #1759]

5466.	[bug]		Addressed an error in recursive clients stats reporting.
			[GL #1719]

5465.	[func]		Added fallback to built-in trust-anchors, managed-keys,
			or trusted-keys if the bindkeys-file (bind.keys) cannot
			be parsed. [GL #1235]

5464.	[bug]		Requesting more than 128 files to be saved when rolling
			dnstap log files caused a buffer overflow. This has been
			fixed. [GL #1989]

5463.	[placeholder]

5462.	[bug]		Move LMDB locking from LMDB itself to named. [GL #1976]

5461.	[bug]		The STALE rdataset header attribute was updated while
			the write lock was not being held, leading to incorrect
			statistics. The header attributes are now converted to
			use atomic operations. [GL #1475]

5460.	[cleanup]	tsig-keygen was previously an alias for
			ddns-confgen and was documented in the ddns-confgen
			man page. This has been reversed; tsig-keygen is
			now the primary name. [GL #1998]


bind-contrib/contrib/updatehosts-1.1.0/readinfo_pp.pl
#! @PERL@

# $Id: readinfo_pp.pl,v 1.1 2000/07/11 06:53:10 vixie Exp $

undefine(`shift')
undefine(`format')
include(`parse_line.pl')

# pretty print a readinfo input file

@line = <>;
chomp(@line);

# sentinel at the end
push(@line, "#FIELDS");
$start = -1;
$line_num = -1;
for $line (@line) {
	$line_num++;
	if ($line =~ /^#FIELDS/) {
		$end = $line_num - 1;
		# beginning of file
		if ($start == -1) {
			$start = 0;
			next;
		}
		for ($i=$start; $i<=$end; $i++) {
			if ($line[$i] =~ /^\s*#/ || $line[$i] =~ /^\s*$/) {
				print "$line[$i]\n";
				next;
			}
			$n = 0;
			@field = parse_line($line[$i]);
			for $field (@field) {
				$l = $length[$n];
				$format = "%-$l"."s ";
				# XXX add quote checking
				printf $format, $field;
				$n++;
			}
			print "\n";
		}
		@field = ();
		@length = ();
		$start = $line_num;
		next;
	}
	next if $line =~ /^\s*#/;
	@field = parse_line($line);
	$n = 0;
	for $field (@field) {
		$l = length($field);
		$l += 2 if $field =~ /^"/ || $field =~ / /;
		$length[$n] = $l if $l > $length[$n];
		$n++
	}
}
